LamestWebserver/LamestWebserver/Core/Logger.cs

```diff
@@ -29,6 +33,66 @@ public class Logger
         public ELoggingLevel MinimumLoggingLevel = DefaultMinimumLoggingLevel;
 
         /// <summary>
+        /// Set the Flags for the Output Sources
+        /// </summary>
+        public EOutputSource OutputSourceFlags
+        {
+            get
+            {
+                return _currentOutputSource;
+            }
+            set
+            {
+                if (_currentOutputSource == value) return;
+                _currentOutputSource = value;
+                CurrentLogger.Instance.RestartStream();
+            }
+        }
+
+        /// <summary>
+        /// Stream Writer to handle the writing from multiple streams
+        /// </summary>
+        private MultiStreamWriter _multiStreamWriter;
+
+        /// <summary>
+        /// Path for the Logging File
+        /// </summary>
+        public string FilePath
+        {
+            get
+            {
+                return _currentFilePath;
+            }
+            set
+            {
+                if (_currentFilePath == value) return;
+                _currentFilePath = value;
+                RestartStream();
+            }
+        }
+        /// <summary>
+        /// Currently used File Path
+        /// </summary>
+        private string _currentFilePath = "LWS" + DateTime.Now.ToFileTime().ToString() + ".log";
+
+        private UsableWriteLock _loggerWriteLock = new UsableWriteLock();
+
+        private ActionList<Stream> _streams = new ActionList<Stream>(() =>
+        {
```

Comment:
Eventhough I'm a big fan of having an `ActionList<T>` I'd really prefer to just manually call `RestartStream()` whenever you're adding, removing or clearing. This `ActionList` increases the whole level of complexity of this class, without adding further functionality. (or am I missing something)

imo the whole chain of redirection in here is way to deep:
```
AddCustomStream
_streams.Add
RestartStream
Close
Open
_createStreamWriter
// actual logic here
```

Reply:
especially since `_streams.Add` internally calls a delegate which calls `RestartStream` which is complex for someone who's new to the codebase.

Reply:
what do you think?

Reply:
but please do not delete the `ActionList<T>` class. We definitely want to keep the class!